krystex/src/main/java/com/flipkart/krystal/krystex/node/Node.java

```diff
@@ -274,12 +305,17 @@ private void executeResolver(RequestId requestId, ResolverDefinition resolverDef
     NodeId depNodeId = nodeDefinition.dependencyNodes().get(dependencyName);
     Inputs inputsForResolver = getInputsForResolver(resolverDefinition, requestId);
```

Comment:
This used only in the else block. So let's move this into the else block. This will avoid the need for putting the skipNodeRequested check in the getInputsForResolver method as well